### PR TITLE
feat(connector): [Bitpay] Use `connector_request_reference_id` as reference to the connector

### PR DESCRIPTION
**Pull request by Himanshu-370**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Make any Payment for connector Bitpay and see that you are getting "reference_id" field in the logs of payment request.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by Himanshu-370** [comment]
image


How to solve this?

**Comment by SanchithHegde** [comment]
@Himanshu-370 You'll have to install `vcpkg` on your machine and then install the OpenSSL package using `vcpkg`.

**Comment by Himanshu-370** [comment]
Is it under review or do I need to solve the CI check fails? cause I am not able to:(

**Comment by deepanshu-iiitu** [comment]
> Is it under review or do I need to solve the CI check fails? cause I am not able to:(
Hi @Himanshu-370 
Please use the following command to generate the openapi spec file
`cargo run --features openapi -- generate-openapi-spec`

File: crates/router/src/connector/bitpay/transformers.rs [hunk 1]
```diff
@@ -138,7 +139,12 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                // connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response
+                        .reference
+                        .unwrap_or(item.response.transaction_id),
+                ),
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/bitpay/transformers.rs:147]: Please revert back these changes as this issue is only about request reference ID. Remove these 6 lines.

Comment (deepanshu-iiitu) [reply crates/router/src/connector/bitpay/transformers.rs:147]: @Himanshu-370

**Comment by swangi-kumari** [comment]
Hey @Himanshu-370 ,
Please address the failing CI checks
Run `cargo clippy` and addressed lints thrown by it.

**Comment by Himanshu-370** [comment]
> Hey @Himanshu-370 , Please address the failing CI checks Run `cargo clippy` and addressed lints thrown by it.

Yes I am trying but I am having trouble installing openssl

**Comment by deepanshu-iiitu** [comment]
> > Hey @Himanshu-370 , Please address the failing CI checks Run `cargo clippy` and addressed lints thrown by it.
> 
> Yes I am trying but I am having trouble installing openssl

Just do the requested code changes along with `cargo +nightly fmt`  and do the commit

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/bitpay/transformers.rs (hunk 1 above)
Comment (Himanshu-370) [reply crates/router/src/connector/bitpay/transformers.rs]: done

File: crates/router/src/connector/bitpay/transformers.rs [hunk 2]
```diff
@@ -138,7 +139,6 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/bitpay/transformers.rs:141]: @Himanshu-370 Please revert back this changes. This line is required.

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Comment by swangi-kumari** [comment]
Hey @Himanshu-370 ,
Thanks for your interest in contributing to hyperswitch.
Let us know if you need any assistance from our end.
Also, even if hacktoberfest is over, we should celebrate open source everyday and we are open for more contributions from you.
We would still be rewarding folks with goodies even if the PR gets merged post hacktoberfest.
May the Source be with you!

File: crates/router/src/connector/bitpay/transformers.rs [hunk 3]
```diff
@@ -279,6 +280,7 @@ fn get_crypto_specific_payment_data(
         ConnectorAuthType::HeaderKey { api_key } => api_key,
         _ => String::default().into(),
     };
+    let order_id = item.connector_request_reference_id.clone();
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/bitpay/transformers.rs:283]:
```suggestion
    let order_id = item.router_data.connector_request_reference_id.clone();
```

Comment (deepanshu-iiitu) [reply crates/router/src/connector/bitpay/transformers.rs:283]: @Himanshu-370 Please do the above change

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/bitpay/transformers.rs (hunk 3 above)
Comment (Himanshu-370) [reply crates/router/src/connector/bitpay/transformers.rs]: done

**Review by AkshayaFoiger — APPROVED** [review]
Looks good!

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @Himanshu-370!

**Comment by deepanshu-iiitu** [comment]
Hey @Himanshu-370  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**